cpp/src/io/json/read_json.cu

```diff
@@ -212,7 +213,7 @@ table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
     return legacy::read_json(sources, reader_opts, stream, mr);
   }
 
-  if (not should_load_whole_source(reader_opts)) {
+  if (reader_opts.get_byte_range_offset() != 0 or reader_opts.get_byte_range_size() != 0) {
```

Comment:
Do we not need a `range_size < source_size` check here?

Reply:
In that case we would allow users to pass giant `byte_range` when reading non-JSONLines files. I'm not sure this is something we want to do, since failing with byte_range + no JSONLines would become less obvious ("but it works sometimes!").
The change here keeps the original behavior.

Reply:
Makes sense, thanks for clarifying!

Reply:
`should_load_whole_source` changed meaning from "is `byte_range` used" to "does `byte_range` have any impact" but the name stayed the same, so I understand the confusion.